client/file_logging.py

```diff
@@ -0,0 +1,138 @@
+import king_phisher.client.plugins as plugins
+import king_phisher.client.gui_utilities as gui_utilities
+
+try:
+	import logging
+except ImportError:
+	has_logger = False # catch standard ImportError and set has_logger to False
+else:
+	has_logger = True # no errors, so logger has been imported
+
+try:
+	import os
+except ImportError:
+	has_os = False #catch standard ImportError and set has_os to False
+else:
+	has_os = True # no errors, so os has been imported
+
+# default values for configuration options
+DEFAULT_LOG_DIR = os.path.expanduser('~/.config/king-phisher/logs')
+DEFAULT_LOG_FILE_SIZE = 10
+DEFAULT_LOG_FILE_COUNT = 2
+
+# minimum integer option values
+MIN_LOG_FILE_SIZE = 1
+MIN_LOG_FILE_COUNT = 1
+
+# logger name and file name values
+LOGGER_NAME = ''
+LOG_FILE_NAME = 'client_log.log'
+
+class Plugin(plugins.ClientPlugin):
+	authors = ['Zach Janice']
+	title = 'Logger'
+	description = """
+	Keep logs of campaign feedback and results. The directory
+	of the logged file(s), the file size (in MB) of the log, and
+	the number of log files kept can be specified. Plugin requires
+	the 'logging' and 'os' packages available for Python.
+	"""
+	req_packages = {
+		'logger': has_logger,
+		'os': has_os
+	}
+	homepage = 'https://github.com/securestate/king-phisher-plugins'
+	options = [
+		plugins.ClientOptionString(
+			'log_dir',
+			'The directory in which to create the log file(s).',
+			default=DEFAULT_LOG_DIR,
+			display_name='Log Directory'
+		),
+		plugins.ClientOptionInteger(
+			'log_file_size',
+			'The maximum size, in megabytes, of a single log file.',
+			default=DEFAULT_LOG_FILE_SIZE,
+			display_name='Log File Size (MB)'
+		),
+		plugins.ClientOptionInteger(
+			'log_file_count',
+			'The number of log files that will be maintained by the plugin.',
+			default=DEFAULT_LOG_FILE_COUNT,
+			display_name='Number of Log Files'
+		)
+	]
+
+	# this is the primary plugin entry point which is executed when the plugin is enabled
+	def initialize(self):
+		log_dir = self.config['log_dir']
+		unsanitary_log_dir_arg = False
+
+		# attempt to create the directory for the log file if it does not exist
+		try:
+			if not os.path.exists(log_dir):
+				os.mkdir(log_dir)
+		except Exception:
+			# revert to default directory
+			unsanitary_log_dir_arg = True
+			self.config['log_dir'] = DEFAULT_LOG_DIR
```

Comment:
this could be moved to a finally block.
Try:
   something
execept Execetion:
  raise the alarm
finally:
 do this